ngi_pipeline/engines/qc_ngi/workflows.py

```diff
@@ -151,17 +193,22 @@ def workflow_fastq_screen(input_files, output_dir, config):
             else:
                 LOG.error('Files passed as list but more than two elements; '
                           'not a read pair? Skipping. ({})'.format(" ".join(elt)))
+                continue
         elif type(elt) is str or type(elt) is unicode:
             cl += " " + elt
         else:
-            LOG.error("Whatchyoo tryin' to get crazy with ese, "
-                      "don'tchyoo know I'm loco? "
-                      "Ignoring your weird input (not a string, not a list).")
+            LOG.error("Ignoring your weird input (not a string, not a list).")
+            continue
         cl_list.append(cl)
-    if not cl_list:
-        raise ValueError("No valid input files passed in; skipping fastq_screen analysis.")
+    if cl_list:
+        # Module loading
+        modules_to_load = get_all_modules_for_workflow("fastq_screen", config)
+        mod_list = [ "module load {}".format(module) for module in modules_to_load ]
+        if mod_list:
+            cl_list = mod_list + cl_list
     else:
-        return cl_list
+        LOG.info("fastq_screen analysis not needed or input files were invalid.")
+    return cl_list
```

Comment:
Yeah I want it to return a list in either case, regardless of if it's empty or populated. I think that's what this does, right? Or I'm a dummy. Which is possible.